src/librustc_mir/transform/validate.rs

```diff
@@ -257,28 +257,17 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
                     Rvalue::Aggregate(box AggregateKind::Array { .. }, _) => {}
                     // All other aggregates must be gone after some phases.
                     Rvalue::Aggregate(box kind, _) => {
-                        match self.mir_phase {
-                            MirPhase::Build
-                            | MirPhase::Const
-                            | MirPhase::ConstPromotion
-                            // Until the drop elaboration phase finishes, we still have aggregates.
-                            | MirPhase::DropLowering => {}
-                            MirPhase::GeneratorLowering => {
-                                // Generators persist until the state machine transformation.
-                                if !matches!(kind, AggregateKind::Generator(..)) {
-                                    self.fail(
-                                        location,
-                                    "generators should not exist after the deaggregator has run".to_string(),
-                                    );
-                                }
+                        // Until the drop elaboration phase finishes, we still have aggregates.
```

Comment:
The comment explains the negation of what is actually checked.
```suggestion
                        // No more aggregates after drop lowering.
```